src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/Parser.java

```diff
@@ -103,7 +103,9 @@ public Parser(JextractTask.ConstantParser constantParser) {
                 }
             });
 
-        return treeMaker.createHeader(tuCursor, decls);
+        Declaration.Scoped rv = treeMaker.createHeader(tuCursor, decls);
+        treeMaker.freeze();
```

Comment:
Does this mean that we can close the index after this?

Reply:
Once we implement the freeze to also remove the Declaration/Cursor dependency, we can close the clang Index.